### PR TITLE
[EH Pyproto] Release preparation

### PR DESCRIPTION
**Pull request by yunhaoling**
- dropping 2.7/3.6
- dropping async including doc
- dropping http proxy/websocket/custom connection endpoint/connection verify
- dropping uamqp
- readme, changelog updates

### CODE REVIEW

**Review by swathipil — APPROVED** [review]
LGTM!

File: sdk/eventhub/azure-eventhub/README.md [hunk 1]
```diff
@@ -19,6 +19,11 @@ The Azure Event Hubs client library allows for publishing and consuming of Azure
 
 _Azure SDK Python packages support for Python 2.7 has ended 01 January 2022. For more information and questions, please refer to https://github.com/Azure/azure-sdk-for-python/issues/20691_
 
+_This version is our first efforts to build an Azure Event Hubs client library based on pure python implemented AMQP stack.
+Features like async support or AmqpOverWebsocket are unavailable for this version which will be added back in future versions,
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/README.md:23]: Maybe make it a bit more explicit that these will be added back by the time we reach a stable release.
Maybe something like:
```
Features like async support or AmqpOverWebsocket are unavailable for this version which will be added back in future previews as we work towards a stable release.
```

File: sdk/eventhub/azure-eventhub/setup.py [hunk 2]
```diff
@@ -55,7 +55,7 @@
     author_email='azpysdkhelp@microsoft.com',
     url='https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/eventhub/azure-eventhub',
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/setup.py:56]: Do we need to update the URL to point to the branch? Or will the change the URL for the GA EH release as well...?

File: sdk/eventhub/azure-eventhub/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,20 @@
 # Release History
 
+## 5.8.0a1 (Unreleased)
+
+This version and all future versions will require Python 3.7+. Python 2.7 and 3.6 are no longer supported.
```
Comment (yunhaoling) [inline sdk/eventhub/azure-eventhub/CHANGELOG.md:5]: 5.7.0 already includes it
```suggestion
```

File: sdk/eventhub/azure-eventhub/setup.py (hunk 2 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/setup.py:56]: I remember the doc guideline is to point to the main branch links even within a feature branch.

File: sdk/eventhub/azure-eventhub/README.md (hunk 1 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/README.md]: updated!

File: eng/tox/allowed_pylint_failures.py [hunk 4]
```diff
@@ -60,5 +60,6 @@
     "azure-messaging-nspkg",
     "azure-agrifood-farming",
     "azure-ai-language-questionanswering",
-    "azure-ai-language-conversations"
+    "azure-ai-language-conversations",
+    "azure-eventhub"
```
Comment (annatisch) [inline eng/tox/allowed_pylint_failures.py:64]: Could we open an issue to re-enable any analysis/test passes we've opted out of?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py [hunk 5]
```diff
@@ -142,7 +142,7 @@ def _open_with_retry(self):
         # type: () -> None
         return self._do_retryable_operation(self._open, operation_need_param=False)
 
-    def _send_event_data(self, timeout_time=None, last_exception=None):
+    def _send_event_data(self, timeout_time=None):
         # type: (Optional[float], Optional[Exception]) -> None
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py:146]: Type hint hasn't been updated to match?

File: eng/tox/allowed_pylint_failures.py (hunk 4 above)
Comment (yunhaoling) [reply eng/tox/allowed_pylint_failures.py:64]: issue created: https://github.com/Azure/azure-sdk-for-python/issues/22455

**Review by annatisch — APPROVED** [review]

**Comment by yunhaoling** [comment]
hey @scbedd , could you help sanity check that the changes I made related to CI would not affect our capability to publish azure-eventhub from this branch?

**Pull request merged**